modules/tinymce/src/themes/silver/test/ts/browser/editor/toolbar/MultipleToolbarVisibilityTest.ts

```diff
@@ -0,0 +1,43 @@
+import { Waiter } from '@ephox/agar';
+import { Assert, describe, it } from '@ephox/bedrock-client';
+import { Css } from '@ephox/sugar';
+import { McEditor, TinyDom } from '@ephox/wrap-mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+
+describe('browser.tinymce.themes.silver.editor.toolbar.MultipleToolbarVisibilityTest', () => {
+
+  const settings = {
+    inline: true,
+    menubar: false,
+    toolbar: 'bold italic underline | alignleft aligncenter alignright alignjustify',
+    base_url: '/project/tinymce/js/tinymce'
+  };
+
+  it('TINY-8503: Does not leave two toolbars showing', async () => {
+    const editorOne = await McEditor.pFromSettings<Editor>(settings);
+    const editorTwo = await McEditor.pFromSettings<Editor>(settings);
+    editorOne.setContent('<p id="number1"><strong>blarg</strong></p>');
+    editorTwo.setContent('<p id="number2">blarg</p>');
+    editorOne.focus();
+    editorTwo.focus();
+    await Waiter.pWait(500);
```

Comment:
This is a very long wait and something we normally discourage... why exactly do we need this?

Reply:
I should add the normal preference is to wait for some state to change. A timeout should only be used where nothing changes and in that case it should be as small as possible.

Reply:
The timeout was 100 initially but the tests were flaky. I can try another approach to avoid using timeout

Reply:
The problem is we're waiting for the bug to _not_ happen. The timeout that triggered the bug is removed by this PR.

Reply:
I guess the other approach is to wait for the editor's`focus` event (if there is) and then timeout a bit before calling assertions. Might be less flaky even with a small timeout like 100.

Reply:
Yeah, from what I can see we can wait for the editor UI to be rendered first. We will likely still need a small delay to ensure nothing renders after but it would be less arbitrary and shouldn't need to be as large.

Reply:
@TheSpyder, `timeout`s are replaced with waiting for `focus` event promises. They are no longer flaky and fail if the fix is removed.

Reply:
As long as it fails with the fix removed (and you can see two toolbars in the failure) I'm happy 👍